tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -0,0 +1,213 @@
+                return
+
+        rsp = self.admin.get_enterprise_features().json()
+
+        expect_status = EnterpriseLicenseStatus.valid if with_license else EnterpriseLicenseStatus.not_present
+        status = rsp.get('license_status', None)
+        assert type(status) == str, f"Ill-formed license_status {type(status)}"
+        try:
+            assert EnterpriseLicenseStatus(
+                status) == expect_status, f"Unexpected status '{status}'"
+        except ValueError:
+            assert False, f"Unexpected status in response: '{status}'"
+
+        violation = rsp.get('violation', None)
+        assert type(
+            violation) == bool, f"Ill-formed violation flag {type(violation)}"
+        assert not violation, "Config unexpectedly in violation"
+
+        features_rsp = [f['name'] for f in rsp.get('features', [])]
+
+        assert set([f.name for f in Features]) == set(
+            features_rsp
+        ), f"Unexpected feature list: {json.dumps(features_rsp, indent=1)}"
+
+    @cluster(num_nodes=3)
+    @matrix(feature=[f for f in Features if f not in SKIP_FEATURES],
+            with_license=[
+                True,
+                False,
+            ])
+    def test_license_violation(self, feature, with_license):
+        if with_license:
+            lic = self._put_license()
+            if lic is None:
+                self.logger.info(
+                    "Skipping test, REDPANDA_SAMPLE_LICENSE env var not found")
+                return
+
+        if feature == Features.audit_logging:
+            self.redpanda.set_cluster_config(
+                {
+                    'audit_enabled': True,
+                },
+                expect_restart=True,
+            )
+        elif feature == Features.cloud_storage:
+            self.redpanda.set_cluster_config({'cloud_storage_enabled': 'true'},
+                                             expect_restart=True)
+        elif feature == Features.partition_auto_balancing_continuous:
+            self.redpanda.set_cluster_config(
+                {'partition_autobalancing_mode': 'continuous'})
+        elif feature == Features.core_balancing_continuous:
+            self.redpanda.set_cluster_config(
+                {'core_balancing_continuous': 'true'})
+        elif feature == Features.gssapi:
+            self.redpanda.set_cluster_config(
+                {'sasl_mechanisms': ['SCRAM', 'GSSAPI']})
+        elif feature == Features.oidc:
+            self.redpanda.set_cluster_config(
+                {'sasl_mechanisms': ['SCRAM', 'OAUTHBEARER']})
+        elif feature == Features.schema_id_validation:
+            self.redpanda.set_cluster_config(
+                {'enable_schema_id_validation': 'compat'})
+        elif feature == Features.rbac:
+
+            # NOTE(oren): make sure the role has propagated to every node since we don't know
+            # where the get_enterprise request will go
```

Comment:
Do we need a similar wait_until for cluster configs?

Reply:
I assumed we were covered by `RedpandaService._wait_for_config_version`, but glancing over the source I'm not so sure. I've run these many times w/o any flake, but I'll look into it.

Reply:
I think we're fine here. syncing on config version seems to be enough

Reply:
Nice, I didn't know `set_cluster_config` did that under the hood